static/app/views/dashboards/utils.tsx

```diff
@@ -468,8 +468,7 @@ export function isWidgetUsingTransactionName(widget: Widget) {
 
 export function hasSavedPageFilters(dashboard: DashboardDetails) {
   return !(
-    dashboard.projects &&
-    dashboard.projects.length === 0 &&
+    (dashboard.projects === undefined || dashboard.projects.length === 0) &&
```

Comment:
cc @narsaynorath idk if you remember what this does

Reply:
For dashboards, instead of serializing all of the page filter values in the list request for the list page, we only get them when you load the dashboard specifically, and then we'll push the saved filters to the URL if there are any. 

Thanks for catching this. I think the case where `projects` being undefined was missed because we have one dashboard that was historically returned in an API request, and that wasn't updated to enforce an empty array for this condition to terminate properly